census/tftransformestimator/trainer/model.py

```diff
@@ -0,0 +1,293 @@
+  transformed_feature_spec = transformed_metadata.schema.as_feature_spec()
+
+  def parser(record):
+    """Help function to parse tf.Example"""
+    parsed = tf.parse_single_example(record, transformed_feature_spec)
+    label = parsed.pop(LABEL_KEY)
+    return parsed, label
+
+  def input_fn():
+    """Input function for training and eval."""
+    #See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/docs_src/performance/datasets_performance.md
+    files = tf.data.Dataset.list_files(os.path.join(tft_working_dir, filebase + '*'))
+    dataset = files.interleave(lambda x : tf.data.TFRecordDataset(x).map(parser,
+                                                                         num_parallel_calls=2),
+                                  cycle_length=4,
+                                  block_length=16
+                               )
+    if shuffle:
+        dataset = dataset.shuffle(buffer_size)
+
+    dataset = dataset.repeat(num_epochs)
+    dataset = dataset.batch(batch_size)
+
+    dataset = dataset.prefetch(prefetch_buffer_size)
+    iterator = dataset.make_one_shot_iterator()
+    transformed_features, transformed_labels = iterator.get_next()
+
+    return transformed_features, transformed_labels
+
+  return input_fn
+
+
+def _get_transformed_features(tft_working_dir):
+  """Return the transformed features
+
+  Args:
+    tft_working_dir : Directory pointed from the tf transform pipeline
+
+  Returns:
+    A dictionary of the transformed features
+  """
+  raw_feature_spec = RAW_DATA_METADATA.schema.as_feature_spec()
+  raw_feature_spec.pop(LABEL_KEY)
+
+  raw_input_fn = tf.estimator.export.build_parsing_serving_input_receiver_fn(
+      raw_feature_spec)
+  raw_features, _ , _ = raw_input_fn()
+
+  # Apply the transform function that was used to generate the materialized
+  # data.
+  _, transformed_features = (
+      saved_transform_io.partially_apply_saved_transform(
+          os.path.join(tft_working_dir, transform_fn_io.TRANSFORM_FN_DIR),
+          raw_features))
+
+  return transformed_features
+
+
+def _make_serving_input_fn(tft_working_dir):
+  """Creates an input function from serving.
+
+  Args:
+    tft_working_dir: Directory to read transformed data and metadata from and to
+        write exported model to.
+
+  Returns:
+    The input function for serving.
+  """
+  raw_feature_spec = RAW_DATA_METADATA.schema.as_feature_spec()
+
+
+  def input_fn():
+    """Serving input function that reads raw data and applies transforms."""
+    raw_placeholder_spec={}
+
+    raw_placeholder_spec = RAW_DATA_METADATA.schema.as_batched_placeholders()
+    # remove label key that is not going to be available at seving
+    raw_placeholder_spec.pop(LABEL_KEY)
+
+    raw_input_fn = tf.estimator.export.build_raw_serving_input_receiver_fn(
+        raw_placeholder_spec)
+    raw_features, receiver_tensors , _ = raw_input_fn()
+
+    _, transformed_features = (
+        saved_transform_io.partially_apply_saved_transform(
```

Comment:
This seems to be the counterpart of your comments on preprocessing.py line 172-174, where transform_fn was written to GCS.  Please add some comments here to highlight that.